### PR TITLE
[Merged by Bors] - feat(CategoryTheory): triangulated functors are additive

### PR DESCRIPTION
**Pull request by joelriou**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by dagurtomas — APPROVED** [review]
LGTM

File: Mathlib/CategoryTheory/Triangulated/Functor.lean [hunk 1]
```diff
@@ -162,6 +162,34 @@ instance (priority := 100) : PreservesZeroMorphisms F where
...
+            ← F.map_comp, F.map_zero]
+      comm₂ := by simp
+      comm₃ := by simp }
+  exact isIso₂_of_isIso₁₃ φ (F.map_distinguished _ (binaryProductTriangle_distinguished X₁ X₃))
+    (binaryProductTriangle_distinguished _ _)
+    (by dsimp; infer_instance) (by dsimp; infer_instance)
+
+instance (priority := 100) : F.Additive := F.additive_of_preserves_binary_products
+
+
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Triangulated/Functor.lean:192]:
```suggestion

```

File: Mathlib/CategoryTheory/Triangulated/Functor.lean [hunk 2]
```diff
@@ -162,6 +162,32 @@ instance (priority := 100) : PreservesZeroMorphisms F where
...
+            limit.lift_π, BinaryFan.mk_pt, BinaryFan.π_app_right, BinaryFan.mk_snd,
+            ← F.map_comp, F.map_zero]
+      comm₂ := by simp
+      comm₃ := by simp }
+  exact isIso₂_of_isIso₁₃ φ (F.map_distinguished _ (binaryProductTriangle_distinguished X₁ X₃))
+    (binaryProductTriangle_distinguished _ _)
+    (by dsimp; infer_instance) (by dsimp; infer_instance)
+
+instance (priority := 100) : F.Additive := F.additive_of_preserves_binary_products
 end IsTriangulated
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Triangulated/Functor.lean:191]:
```suggestion

end IsTriangulated
```

**Review by TwoFX — APPROVED** [review]
Thanks!
bors r+

**Pull request closed without merging**